### PR TITLE
Fix multilevel optionals.

### PR DESCRIPTION
**Pull request by Tony-Romanov**
Type: fix
Component: query-tracker

### CODE REVIEW

File: yt/yql/tests/agent/test_simple.py [hunk 1]
```diff
@@ -228,6 +228,13 @@ def test_exotic_types(self, query_tracker, yql_agent):
                "Yson": [7, 'str', -3.14]
                }])
 
+    @authors("a-romanov")
+    def test_double_optional(self, query_tracker, yql_agent):
```
Comment (Krock21) [inline yt/yql/tests/agent/test_simple.py:232]: Could you please add a more complex test with combination of inner maps and lists, like `[[[null],[null]],[[null],[null]]]`

To test logic with OptionalLevels and Depth

**Review by nepal — APPROVED** [review]

File: yt/yql/agent/data_builder.cpp [hunk 2]
```diff
@@ -342,6 +359,18 @@ void TDataBuilder::OnPg(TMaybe<TStringBuf> value, bool /*isUtf8*/)
     }
 }
 
+void TDataBuilder::AddBeginOptional() {
```
Comment (Krisha11) [inline yt/yql/agent/data_builder.cpp:362]: { 
should be on the next line

File: yt/yql/agent/data_builder.cpp [hunk 3]
```diff
@@ -342,6 +359,18 @@ void TDataBuilder::OnPg(TMaybe<TStringBuf> value, bool /*isUtf8*/)
...
+    if (OptionalLevels_.top() > 1) {
+        BeginList();
+    }
+}
+
+void TDataBuilder::AddEndOptional() {
```
Comment (Krisha11) [inline yt/yql/agent/data_builder.cpp:368]: {

**Review by Krisha11 — APPROVED** [review]
